frontend/ui/widget/textviewer.lua

```diff
@@ -441,6 +441,7 @@ function TextViewer:findDialog()
             {
                 {
                     text = _("Cancel"),
+                    id = "close",
```

Comment:
Remind me why the need for all these added `id="close"` ?

Reply:
ButtonDialog is closed by tapping on its title ("inside").

Reply:
id is used for non-touch devices and in
https://github.com/koreader/koreader/blob/082ef9b545809f865e1eeeaa8e9733a20120a1ad/frontend/ui/widget/inputdialog.lua#L635-L642

Reply:
> ButtonDialog is closed by tapping on its title ("inside"). (reply to the previous comment)

Just tested that (eg. with the dialog we get on long-press on a file in History), and it also feels odd :)

Reply:
So, your proposition is
https://github.com/koreader/koreader/blob/082ef9b545809f865e1eeeaa8e9733a20120a1ad/frontend/ui/widget/confirmbox.lua#L245-L251

Reply:
Or even tapping outside should not close the InputDialog? (to not lose the text)

Reply:
Yes, that's something like that I was expecting to read.